cpu/stm32_common/clk/pll.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *               2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_stm32_common
+ * @{
+ *
+ * @file
+ * @brief       STM32 main PLL configuration
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @}
+ */
+
+#ifndef PLL_H
+#define PLL_H
+
+/* the recommended input clock for the PLL should be 2MHz */
+#ifndef PLL_IN_FREQ
+#if ((PLL_IN / 2) * 2 == PLL_IN)
+#define PLL_IN_FREQ              (2000000U)
+#else
+#define PLL_IN_FREQ              (1000000U)
+#endif
+#endif
+#define M                        (PLL_IN / PLL_IN_FREQ)
+#if ((M < 2) || (M > 63))
+#error "PLL configuration: PLL M value is out of range"
+#endif
+#if (PLL_IN_FREQ * M) != PLL_IN
+#error "PLL configuration: PLL input frequency is invalid (M)"
+#endif
+
+/* Use user-provided P if existing, otherwise try possible values */
+#ifdef P
+#if ((P != 2) && (P != 4) && (P != 6) && (P != 8))
+#error "PLL configuration: PLL P value is invalid"
+#endif
+#define N                       (P * CLOCK_CORECLOCK / PLL_IN_FREQ)
+#if ((N < 50) || (N > 432))
+#error "PLL configuration: PLL N value is out of range"
+#endif
+#if (PLL_IN_FREQ * N / P) != CLOCK_CORECLOCK
+#error "PLL configuration: PLL input frequency is invalid (N)"
+#endif
+#else
+#define P  (2U)
+#define N  (P * CLOCK_CORECLOCK / PLL_IN_FREQ)
+#if (N >= 50) && (N <= 432) && ((PLL_IN_FREQ * N / P) == CLOCK_CORECLOCK)
+#define _PLL_N_FINISHED 1
+#endif
+#if !_PLL_N_FINISHED
+#undef P
```

Comment:
yes because `P` could be already defined above, which would cause a preprocessor error on the (re)define below.